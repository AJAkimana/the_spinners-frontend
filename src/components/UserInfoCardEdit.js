import 'date-fns';
import React from 'react';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import Button from '@material-ui/core/Button';
import { Divider, makeStyles, Typography } from '@material-ui/core';
import CardHeader from '@material-ui/core/CardHeader';
import TextField from '@material-ui/core/TextField';
import Radio from '@material-ui/core/Radio';
import DateFnsUtils from '@date-io/date-fns';
import { useDispatch } from 'react-redux';
import {
  MuiPickersUtilsProvider,
  KeyboardDatePicker
} from '@material-ui/pickers';
import { updateProfile } from '../redux/actions/updateProfileActions'

const useStyles = makeStyles(theme => ({
  root: {
    '& .MuiTextField-root': {
      margin: theme.spacing(1)
    },
    borderRadius: 0
  },
  header: {
    textAlign: 'center',
    backgroundColor: '#0074D9',
    color: '#ffffff'
  },
  typos: {
    fontWeight: 'bold',
    fontSize: '16px',
    paddingRight: '5px'
  },
  submitInfoBtn: {
    textAlign: 'center'
  },
  button: {
    backgroundColor: '#0074D9',
    '&:hover': {
      backgroundColor: '#0069d9'
    },
    '&:focus': {
      boxShadow: '0 0 0 0.2rem rgba(0,123,255,.5)',
    },
    fontSize: '10px',
    fontWeight: 'bold',
    borderRadius: 4,
    marginTop: 30,
    padding: 'auto'
  },
  action: {
    display: 'flex',
    justifyContent: 'space-around',
  },
  userInfoValues: {
    fontSize: '14px',
  },
  contents: {
    maxHeight: 'auto'
  },
  formInput: {
    width: '70%',
  },
  formDiv: {
    width: '100%',
  },
  GenderParagraph: {
    paddingLeft: '50px'
  }
}));

export default function UserInfoCardEdit() {
  const classes = useStyles();
  const [residence, setResidence] = React.useState('Kigali');
  const handleResidenceChange = event => {
    setResidence(event.target.value);
  };

  const [gender, setGender] = React.useState('M');
  const handleGenderChange = event => {
    setGender(event.target.value);
  };

  const [language, setLanguage] = React.useState('english');
  const handleLanguageChange = event => {
    setLanguage(event.target.value);
  }

  const [currency, setCurrency] = React.useState('Dollar');
  const handleCurrencyChange = event => {
    setCurrency(event.target.value);
  }

  const [birthDate, setbirthDate] = React.useState(new Date('1990-01-01'));
  const handleDateChange = event => {
    setbirthDate(event.target.value);
  };

  const dispatch = useDispatch();
  const handleSubmit = (e) => {
    e.preventDefault();
    dispatch(updateProfile(
      {
        gender,
        residence,
        preferredLanguage: language,
        preferredCurrency: currency,
        // birthDate
      }))
  }

  return (
    <Card className={classes.root}>
      <CardHeader title="Personal Information" className={classes.header} />
      <Divider variant="middle" />
      <CardContent className={classes.contents}>
        <form onSubmit={handleSubmit} className={classes.root} noValidate autoComplete="off">
          <div className={classes.formDiv}>
            <div style={{ marginLeft: 45 }}>
              <TextField
                className={classes.formInput}
                id="residence"
                label="Residence"
                variant="outlined"
                size="small"
                fullWidth
                onChange={handleResidenceChange}
              />
            </div>
            <br />
            <div className={classes.GenderParagraph}>
              <Typography style={{ fontWeight: 'bold', textDecoration: 'underline', fontSize: 13 }}>
                Gender
              </Typography>
              <Radio
                checked={gender === 'M'}
                value="M"
                color="default"
                name="radio-button-demo"
                inputProps={{ 'aria-label': 'M' }}
                onChange={handleGenderChange}
              />
              Male
              <Radio
                checked={gender === 'F'}
                value="F"
                color="default"
                name="radio-button-demo"
                inputProps={{ 'aria-label': 'F' }}
                onChange={handleGenderChange}
              />
              Female
            </div>
            <br />
            <div style={{ paddingLeft: '50px' }}>
              <Typography style={{ fontWeight: 'bold', textDecoration: 'underline', fontSize: 13 }}>
                Language
              </Typography>
              <Radio
                checked={language === 'english'}
                onChange={handleLanguageChange}
                value="english"
                color="default"
                name="radio-button-demo"
                inputProps={{ 'aria-label': 'english' }}
              />
              English
              <Radio
                checked={language === 'french'}
                onChange={handleLanguageChange}
                value="french"
                color="default"
                name="radio-button-demo"
                inputProps={{ 'aria-label': 'french' }}
              />
              French
            </div>
            <br />
            <div style={{ paddingLeft: '50px' }}>
              <Typography style={{ fontWeight: 'bold', textDecoration: 'underline', fontSize: 13 }}>
                Currency
              </Typography>
              <Radio
                checked={currency === 'Euro'}
                onChange={handleCurrencyChange}
                value="Euro"
                color="default"
                name="radio-button-demo"
                inputProps={{ 'aria-label': 'Euro' }}
              />
              Euro
              <Radio
                checked={currency === 'Dollar'}
                onChange={handleCurrencyChange}
                value="Dollar"
                color="default"
                name="radio-button-demo"
                inputProps={{ 'aria-label': 'Dollar' }}
              />
              US Dollar
              <Radio
                checked={currency === 'Pound'}
                onChange={handleCurrencyChange}
                value="Pound"
                color="default"
                name="radio-button-demo"
                inputProps={{ 'aria-label': 'Pound' }}
              />
              Pound
            </div>
            <br />
            <div style={{ paddingLeft: '50px' }}>
              <Typography style={{ fontWeight: 'bold', textDecoration: 'underline', fontSize: 13 }}>
                Birth Date
              </Typography>
              <MuiPickersUtilsProvider utils={DateFnsUtils}>
                <KeyboardDatePicker
                  disableToolbar
                  variant="inline"
                  format="yyyy-MM-dd"
                  margin="normal"
                  id="Birth date"
                  value={birthDate}
                  onChange={handleDateChange}
                  KeyboardButtonProps={{
                    'aria-label': 'change date',
                  }}
                />
              </MuiPickersUtilsProvider>
            </div>
          </div>
          <div className={classes.submitInfoBtn}>
            <Button type="submit" variant="contained" color="primary" className={classes.button}>
              update
            </Button>
          </div>
        </form>
      </CardContent>
    </Card>
  );
};
