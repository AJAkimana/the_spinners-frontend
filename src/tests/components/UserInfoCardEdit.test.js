/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import UserInfoCardEdit from '../../components/UserInfoCardEdit';

jest.mock('react-redux', () => ({
	useSelector: jest.fn(() => ({
		userData: {
			firstName: 'Gustave',
			lastName: 'Harintwari',
			email: 'higustave@gmail.com',
			gender: 'M',
			language: 'english',
			currency: 'Euro',
			residence: 'Rwanda',
			profilePicture: 'dgshajskmjhjnbdsa',
		},
	})),
	useDispatch: jest.fn(),
}));

describe('Test On UserInfoCardEdit Component', () => {
	const wrapper = shallow(<UserInfoCardEdit />);
	const component = wrapper.dive();

	describe('render()', () => {
		test('Should Render UserInfoCardEdit Component', () => {
			expect(toJson(component)).toMatchSnapshot();
		});
	});
	describe('onChange()', () => {
		test('Should Successfully call handleGenderChange Method', () => {
			wrapper.find('.gender').simulate('change', { target: { value: 'M' } });
		});
		test('Should Successfully call handleLanguageChange Method', () => {
			wrapper.find('.language').simulate('change', { target: { value: 'english' } });
		});
		test('Should Successfully call handleCurrencyChange Method', () => {
			wrapper.find('.currency').simulate('change', { target: { value: 'Euro' } });
		});
		test('Should Successfully call handleBirthDateChange Method', () => {
			wrapper.find('.birthDate').simulate('change', { target: { value: '1993-07-29' } });
		});
	});
});
