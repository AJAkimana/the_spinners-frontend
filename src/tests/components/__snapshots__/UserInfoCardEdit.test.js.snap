// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test On UserInfoCardEdit Component render() Should Render UserInfoCardEdit Component 1`] = `
<ForwardRef(Card)
  className="makeStyles-root-1"
  classes={
    Object {
      "root": "MuiCard-root",
    }
  }
>
  <WithStyles(ForwardRef(CardContent))
    className="makeStyles-contents-8"
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <form
      autoComplete="off"
      className="makeStyles-root-1"
      noValidate={true}
      onSubmit={[Function]}
    >
      <div
        className="makeStyles-formDiv-10"
      >
        <div
          style={
            Object {
              "display": "flex",
              "textAlign": "center",
            }
          }
        >
          <div
            style={
              Object {
                "margin": "auto",
                "width": "500px",
              }
            }
          >
            <WithStyles(ForwardRef(TextField))
              className="makeStyles-formInput-9"
              disabled={true}
              id="firstname"
              label="First name"
              size="small"
              variant="outlined"
            />
          </div>
        </div>
        <div
          style={
            Object {
              "display": "flex",
              "textAlign": "center",
            }
          }
        >
          <div
            style={
              Object {
                "margin": "auto",
                "width": "500px",
              }
            }
          >
            <WithStyles(ForwardRef(TextField))
              className="makeStyles-formInput-9"
              disabled={true}
              id="lastname"
              label="Last name"
              size="small"
              variant="outlined"
            />
          </div>
        </div>
        <div
          style={
            Object {
              "display": "flex",
              "textAlign": "center",
            }
          }
        >
          <div
            style={
              Object {
                "margin": "auto",
                "width": "500px",
              }
            }
          >
            <WithStyles(ForwardRef(TextField))
              className="makeStyles-formInput-9"
              disabled={true}
              id="email"
              label="Email"
              size="small"
              variant="outlined"
            />
          </div>
        </div>
        <div
          style={
            Object {
              "display": "flex",
              "textAlign": "center",
            }
          }
        >
          <div
            style={
              Object {
                "margin": "auto",
                "width": "500px",
              }
            }
          >
            <WithStyles(ForwardRef(FormControl))
              className="makeStyles-formControl-12"
              size="small"
              variant="outlined"
            >
              <WithStyles(ForwardRef(InputLabel))
                htmlFor="outlined-gender-native-simple"
              >
                Gender
              </WithStyles(ForwardRef(InputLabel))>
              <WithStyles(ForwardRef(Select))
                className="gender"
                inputProps={
                  Object {
                    "id": "outlined-gender-native-simple",
                    "name": "gender",
                  }
                }
                label="gender-input"
                labelWidth={0}
                native={true}
                onChange={[Function]}
              >
                <option
                  value=""
                />
                <option
                  value="M"
                >
                  Male
                </option>
                <option
                  value="F"
                >
                  Female
                </option>
              </WithStyles(ForwardRef(Select))>
            </WithStyles(ForwardRef(FormControl))>
          </div>
        </div>
        <div
          style={
            Object {
              "display": "flex",
              "textAlign": "center",
            }
          }
        >
          <div
            style={
              Object {
                "margin": "auto",
                "width": "500px",
              }
            }
          >
            <WithStyles(ForwardRef(FormControl))
              className="makeStyles-formControl-12"
              size="small"
              variant="outlined"
            >
              <WithStyles(ForwardRef(InputLabel))
                htmlFor="outlined-language-native-simple"
              >
                Language
              </WithStyles(ForwardRef(InputLabel))>
              <WithStyles(ForwardRef(Select))
                className="language"
                inputProps={
                  Object {
                    "id": "outlined-language-native-simple",
                    "name": "language",
                  }
                }
                labelWidth={0}
                native={true}
                onChange={[Function]}
              >
                <option
                  value=""
                />
                <option
                  value="english"
                >
                  English
                </option>
                <option
                  value="french"
                >
                  French
                </option>
              </WithStyles(ForwardRef(Select))>
            </WithStyles(ForwardRef(FormControl))>
          </div>
        </div>
        <div
          style={
            Object {
              "display": "flex",
              "textAlign": "center",
            }
          }
        >
          <div
            style={
              Object {
                "margin": "auto",
                "width": "500px",
              }
            }
          >
            <WithStyles(ForwardRef(FormControl))
              className="makeStyles-formControl-12"
              size="small"
              variant="outlined"
            >
              <WithStyles(ForwardRef(InputLabel))
                htmlFor="outlined-currency-native-simple"
              >
                Currency
              </WithStyles(ForwardRef(InputLabel))>
              <WithStyles(ForwardRef(Select))
                className="currency"
                inputProps={
                  Object {
                    "id": "outlined-currency-native-simple",
                    "name": "currency",
                  }
                }
                labelWidth={0}
                native={true}
                onChange={[Function]}
              >
                <option
                  value=""
                />
                <option
                  value="Dollar"
                >
                  USD $
                </option>
                <option
                  value="Euro"
                >
                  Euro €
                </option>
                <option
                  value="Pound"
                >
                  Pound £
                </option>
              </WithStyles(ForwardRef(Select))>
            </WithStyles(ForwardRef(FormControl))>
          </div>
        </div>
        <div
          style={
            Object {
              "display": "flex",
              "textAlign": "center",
            }
          }
        >
          <div
            style={
              Object {
                "margin": "auto",
                "width": "500px",
              }
            }
          >
            <CountrySelect
              className="countries"
              countryChangeFn={[Function]}
              inputProps={
                Object {
                  "id": "outlined-residence-native-simple",
                  "name": "residence",
                }
              }
            />
          </div>
        </div>
        <div
          style={
            Object {
              "display": "flex",
              "textAlign": "center",
            }
          }
        >
          <div
            className="birthDate"
            onChange={[Function]}
            style={
              Object {
                "margin": "auto",
                "width": "500px",
              }
            }
          >
            <WithStyles(ForwardRef(TextField))
              InputLabelProps={
                Object {
                  "shrink": true,
                }
              }
              defaultValue="1990-01-01"
              label="Birthday"
              size="small"
              type="date"
            />
          </div>
        </div>
      </div>
      <div
        className="makeStyles-submitInfoBtn-4"
      >
        <WithStyles(ForwardRef(Button))
          className="makeStyles-button-5"
          color="primary"
          disabled={true}
          size="large"
          style={
            Object {
              "fontSize": "11px",
              "height": "40px",
            }
          }
          type="submit"
          variant="contained"
        >
          UPDATE PROFILE
        </WithStyles(ForwardRef(Button))>
      </div>
    </form>
  </WithStyles(ForwardRef(CardContent))>
</ForwardRef(Card)>
`;
